src/Microsoft.Dnx.Tooling/Restore/RestoreCommand.cs

```diff
@@ -225,7 +225,7 @@ public RestoreCommand(FrameworkName fallbackFramework)
             var diagnostics = new List<DiagnosticMessage>();
             if (!Runtime.Project.TryGetProject(projectJsonPath, out project, diagnostics))
             {
-                throw new Exception("TODO: project.json parse error");
+                throw new InvalidOperationException("project.json parse error");
```

Comment:
We should now have error information in the `diagnostics` value. Could you format that (if it's present) and display it? It's possible this could fail but diagnostics would be empty (in which case just use text like "Invalid project.json").

Something like

```
Errors occurred parsing project.json:
project.json([Line#], [Col#]): [Message]
etc.
```

All of that should be available in `diagnostics` and should be pretty straightforward to add with some `string.Joins` and Linq :). Maybe throw it in a helper if it looks like it's going to be a bit complicated.